lib/board.rb

```diff
@@ -0,0 +1,90 @@
+class Board
+  attr_reader :cells, :height
+  def initialize(height = 4, width = 4)
+    @height = height
+    @width = width
+    @cells = generate_cells
+  end
+
+  def board_letters
+    ("A"..("A".ord + @height - 1).chr).to_a
+  end
+
+  def board_numbers
+    ("1"..(@width.to_s)).to_a
+  end
+
+  def generate_cells
+    board_letters.reduce({}) do |generated_cells, letter|
+      board_numbers.each do |number|
+        generated_cells[letter + number] = Cell.new(letter + number)
+      end
+      generated_cells
+    end
+  end
+
+  def valid_placement?(ship, coordinates)
+    valid_coordinate?(coordinates) && valid_length?(ship, coordinates) && !overlapping_coordinates?(coordinates) && consecutive_coordinates?(ship, coordinates)
+  end
+
+  def valid_coordinate?(coordinates)
```

Comment:
Would the name valid_coordinates? with an s be better since this method takes in an array of coordinates?

Reply:
I looked at the valid_coordinate? method in the rubic and it expects to take a string of one coordinate not an array of multiple coordinates.

Reply:
I can see why you need it to be able to accept multiple.

Reply:
Maybe two methods
valid_coordinate?
and 
valid_coordinates?

Reply:
So looking back at this I realized that I had initially skipped over the valid_coordinate? portion in the rubric. When I wrote this method I thought I was super smart and had found an **additional** method I could write for robustness hahaha. 

I works for both single coordinates and multiple as long as they're stored in an array (single would be a one element array). I added tests for this.

The current name matched the rubric so I think we should keep it.